hasktorch/src/Torch/TensorFactories.hs

```diff
@@ -203,155 +247,137 @@ randintIO' low high = mkDefaultFactory (randintIO low high)
-
-rreluIO''
-  :: Scalar a 
-  => Tensor
-  -> a
-  -> IO Tensor
-rreluIO'' t _upper =
-  (cast2 ATen.rrelu_ts) t _upper
-
-rreluIO'
-  :: Scalar a 
-  => Tensor
-  -> a
-  -> a
-  -> IO Tensor
-rreluIO' t _lower _upper =
-  (cast3 ATen.rrelu_tss) t _lower _upper
-
-rreluIO
-  :: Scalar a 
-  => Tensor
-  -> a
-  -> a
-  -> Bool
-  -> IO Tensor
-rreluIO t _lower _upper _training =
-  (cast4 ATen.rrelu_tssb) t _lower _upper _training
-
-rreluWithNoiseIO'''
-  :: Tensor
-  -> Tensor
-  -> IO Tensor
-rreluWithNoiseIO''' t _noise =
-  (cast2 ATen.rrelu_with_noise_tt) t _noise
-
-rreluWithNoiseIO''
-  :: Scalar a 
-  => Tensor
-  -> Tensor
-  -> a
-  -> IO Tensor
-rreluWithNoiseIO'' t _noise _upper =
-  (cast3 ATen.rrelu_with_noise_tts) t _noise _upper
-
-rreluWithNoiseIO'
-  :: Scalar a 
-  => Tensor
-  -> Tensor
-  -> a
-  -> a
-  -> IO Tensor
-rreluWithNoiseIO' t _noise _lower _upper =
-  (cast4 ATen.rrelu_with_noise_ttss) t _noise _lower _upper
-
-rreluWithNoiseIO
-  :: Scalar a 
-  => Tensor
-  -> Tensor
-  -> a
-  -> a
-  -> Bool
-  -> IO Tensor
-rreluWithNoiseIO t _noise _lower _upper _training =
-  (cast5 ATen.rrelu_with_noise_ttssb) t _noise _lower _upper _training
-
-onesWithDimnames' :: [(Int,Dimname)] -> Tensor
+bernoulliIO' ::
+  Tensor ->
+  IO Tensor
+bernoulliIO' = cast1 ATen.bernoulli_t
+
+bernoulliIO ::
+  Tensor ->
+  Double ->
+  IO Tensor
+bernoulliIO = cast2 ATen.bernoulli_td
+
+poissonIO ::
+  Tensor ->
+  IO Tensor
+poissonIO = cast1 ATen.poisson_t
+
+multinomialIO' ::
+  Tensor ->
+  Int ->
+  IO Tensor
+multinomialIO' = cast2 ATen.multinomial_tl
+
+multinomialIO ::
+  Tensor ->
+  Int ->
+  Bool ->
+  IO Tensor
+multinomialIO = cast3 ATen.multinomial_tlb
+
+normalIO' ::
+  Tensor ->
+  IO Tensor
+normalIO' = cast1 ATen.normal_t
+
+normalIO ::
+  Tensor ->
+  Tensor ->
+  IO Tensor
+normalIO = cast2 ATen.normal_tt
+
+normalScalarIO ::
+  Tensor ->
+  Double ->
+  IO Tensor
+normalScalarIO = cast2 ATen.normal_td
+
+normalScalarIO' ::
+  Double ->
+  Tensor ->
+  IO Tensor
+normalScalarIO' = cast2 ATen.normal_dt
+
+normalWithSizeIO ::
+  Double ->
+  Double ->
+  Int ->
+  IO Tensor
+normalWithSizeIO = cast3 ATen.normal_ddl
+
+rreluIO''' ::
+  Tensor ->
+  IO Tensor
+rreluIO''' = cast1 ATen.rrelu_t
+
+rreluIO'' ::
+  Scalar a =>
+  Tensor ->
+  a ->
+  IO Tensor
+rreluIO'' = cast2 ATen.rrelu_ts
+
+rreluIO' ::
+  Scalar a =>
+  Tensor ->
+  a ->
+  a ->
+  IO Tensor
+rreluIO' = cast3 ATen.rrelu_tss
+
+rreluIO ::
+  Scalar a =>
+  Tensor ->
+  a ->
+  a ->
+  Bool ->
+  IO Tensor
+rreluIO = cast4 ATen.rrelu_tssb
+
+rreluWithNoiseIO''' ::
+  Tensor ->
+  Tensor ->
+  IO Tensor
+rreluWithNoiseIO''' = cast2 ATen.rrelu_with_noise_tt
+
+rreluWithNoiseIO'' ::
+  Scalar a =>
+  Tensor ->
+  Tensor ->
+  a ->
+  IO Tensor
+rreluWithNoiseIO'' = cast3 ATen.rrelu_with_noise_tts
+
+rreluWithNoiseIO' ::
+  Scalar a =>
+  Tensor ->
+  Tensor ->
+  a ->
+  a ->
+  IO Tensor
+rreluWithNoiseIO' = cast4 ATen.rrelu_with_noise_ttss
```

Comment:
these are all getting eta-reduced... hmm, it wouldn't be an issue if these arguments were documented :/